File: artifacts/tulipindicators/decompose/src/bop.c
int ti_bop_start(const double *options)
{
  (void) options;
  return 0;
}


/*** DEPENDENCIES:
***/


int ti_bop(int size, const double * const *inputs, const double *options, double * const *outputs)
{
  const double *open = inputs[0];
  const double *high = inputs[1];
  const double *low = inputs[2];
  const double *close = inputs[3];
  (void) options;
  double *output = outputs[0];
  int i;
  for (i = 0; i < size; i += 1)
  {
    double hl = high[i] - low[i];
    if (hl <= 0.0)
    {
      output[i] = 0;
    }
    else
    {
      output[i] = (close[i] - open[i]) / hl;
    }
  }

  return 0;
}


/*** DEPENDENCIES:
***/

File: artifacts/tulipindicators/automatic_debug/src/bop.rs
use crate::*;

pub fn ti_bop_start(options: Option<&[f64]>) -> i32 {
	println!("\n* ti_bop_start");
    let _ = options; // Explicitly ignore the parameter
    0
}
pub fn ti_bop(
    size: usize,
    inputs: &[&[f64]],
    options: &[f64],
    outputs: &mut [&mut [f64]],
) -> i32 {
	println!("\n* ti_bop");
    let open = inputs[0];
    let high = inputs[1];
    let low = inputs[2];
    let close = inputs[3];
    let _ = options; // unused, silence warning
    let output = &mut outputs[0];

    for i in 0..size {
        let hl = high[i] - low[i];
        if hl <= 0.0 {
            output[i] = 0.0;
        } else {
            output[i] = (close[i] - open[i]) / hl;
        }
    }

    0
}
